envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RBACFilterFactory.kt

```diff
@@ -195,7 +198,7 @@ class RBACFilterFactory(
             roles.find { it.name == clientOrRole.name }?.clients ?: setOf(clientOrRole)
         }
         // sorted order ensures that we do not duplicate rules
-        return clients.toSortedSet()
+        return (clients + allowAllEndpointsForClients).toSortedSet()
```

Comment:
I'm starting to get lost in this factory, there are so many conditions and corner cases. I think it's time for a refactor and extracting some patterns from this code.